gspread/worksheet.py

```diff
@@ -160,11 +160,16 @@ def __init__(
         spreadsheet_id: str,
         client: HTTPClient,
         properties: MutableMapping[str, Any],
+        spreadsheet: Any,
     ):
         self.spreadsheet_id = spreadsheet_id
         self.client = client
         self._properties = properties
 
+        # kept for backward compatibility - publicly available
+        # do not use if possible.
+        self.spreadsheet = spreadsheet
```

Comment:
we can make this a private attribute with
_spreadsheet

and then make a method spreadsheet with @property decorator, where we can emit warnings

?

Reply:
you're right that's better, this way we can add some deprecation warnings when needed.